lib/kontena/machine/upcloud/node_provisioner.rb

```diff
@@ -109,9 +109,12 @@ def erb(template, vars)
           ERB.new(template).result(OpenStruct.new(vars).instance_eval { binding })
         end
 
+        # @param [Hash] node
+        # @param [Array<String>] labels
         def set_labels(node, labels)
-          data = {labels: labels}
-          api_client.put("nodes/#{node['grid']['id']}/#{node['id']}", data)
+          data = {}
+          data[:labels] = labels
+          api_client.put("nodes/#{node['id']}", data, {}, {'Kontena-Grid-Token' => node['grid']['token']})
```

Comment:
This should not need the `Kontena-Grid-Token` header if the `api_client` is using the normal oauth bearer token.

Reply:
Yeah, this header does not look right?

Reply:
Strange that it has been there for two years unchanged in AWS plugin: https://github.com/kontena/kontena-plugin-aws/blame/master/lib/kontena/machine/aws/node_provisioner.rb#L201

Reply:
Seems to work without.